plugin-process-to-cgroup-bridge/src/transform.rs

```diff
@@ -0,0 +1,199 @@
+    fn apply(&mut self, measurements: &mut MeasurementBuffer, _ctx: &TransformContext) -> Result<(), TransformError> {
+        let mut cgroup_measurements = MeasurementBuffer::new();
+        let mut kept_measurements = MeasurementBuffer::new();
+        for measurement in measurements.iter() {
+            if let Some(cgroup_measurement) = self.cgroup_measurement_from_measurement(measurement)? {
+                cgroup_measurements.push(cgroup_measurement);
+                if self.keep_processed_measurements {
+                    kept_measurements.push(measurement.clone());
+                }
+            } else {
+                kept_measurements.push(measurement.clone());
+            }
+        }
+        measurements.clear();
+
+        if self.merge_similar_cgroups {
+            let mut aggregated_cgroup_measurements = Self::aggregate_cgroups_measurements(&mut cgroup_measurements);
+            cgroup_measurements.clear();
+            cgroup_measurements.merge(&mut aggregated_cgroup_measurements);
+        }
+
+        measurements.merge(&mut kept_measurements);
+        measurements.merge(&mut cgroup_measurements);
+
+        Ok(())
+    }
+}
+
+impl ProcessToCgroupBridgeTransform {
+    pub fn new(
+        processes_metrics_ids: Vec<RawMetricId>,
+        merge_similar_cgroups: bool,
+        keep_processed_measurements: bool,
+    ) -> Self {
+        Self {
+            processes_metrics_ids,
+            merge_similar_cgroups,
+            keep_processed_measurements,
+        }
+    }
+
+    fn cgroup_measurement_from_measurement(
+        &self,
+        measurement: &MeasurementPoint,
+    ) -> anyhow::Result<Option<MeasurementPoint>> {
+        if !self.processes_metrics_ids.contains(&measurement.metric) {
+            return Ok(None);
+        }
+        let pid = match Self::extract_process_id_from_measurement(measurement) {
+            Ok(pid) => pid,
+            Err(_) => return Ok(None),
+        };
+        match Self::find_cgroup_path_from_process_id(pid)? {
+            Some(cgroup_path) => {
+                let cgroup_consumer = ResourceConsumer::ControlGroup {
+                    path: cgroup_path.into(),
+                };
+                let mut cgroup_measurement = measurement.clone();
+                cgroup_measurement.consumer = cgroup_consumer;
+                Ok(Some(cgroup_measurement))
+            }
+            None => {
+                log::warn!("no cgroup was found for process {pid}");
+                Ok(None)
+            }
+        }
+    }
+
+    fn find_cgroup_path_from_process_id(pid: u32) -> anyhow::Result<Option<String>> {
+        let procfs_cgroup_base_path = get_proc_path();
+
+        let procfs_cgroup_filepath = procfs_cgroup_base_path.join(pid.to_string()).join("cgroup");
+
+        if !Path::new(&procfs_cgroup_filepath).exists() {
+            return Ok(None);
+        }
+
+        let contents = fs::read_to_string(&procfs_cgroup_filepath)
```

Comment:
if the file doesn't exist, it would be more logical to return an error

Reply:
Also, read_to_string already check if it exists. You can detect that by checking the `ErrorKind`. See https://doc.rust-lang.org/std/io/struct.Error.html